corehq/pillows/case.py

```diff
@@ -28,7 +28,14 @@
 pillow_logging.setLevel(logging.INFO)
 
 
+class InvalidChangeException(Exception):
+    pass
+
+
 def transform_case_for_elasticsearch(doc_dict):
+    if doc_dict.get('doc_type') == 'MobileAuthKeyRecord':
+        raise InvalidChangeException("doc_type is not a case")
```

Comment:
This would raise a PillowError, with extra debug info and skip on these bad docs